api/organisations/tasks.py

```diff
@@ -189,7 +197,81 @@ def handle_api_usage_notifications():
             )
 
 
+def restrict_use_due_to_api_limit_grace_period_over() -> None:
```

Comment:
I think I understand what you're asking for here, but just to be clear:

Once this task has been run, but before the individual filtered organisations have been modified we would have to run something like this:

```python
for organisation in organisations:
        stop_serving_flags_enabled = flagsmith_client.get_identity_flags(
            f"org.{organisation.id}.{organisation.name}",
            traits={"organisation_id": organisation.id},
        ).is_feature_enabled("api_limiting_stop_serving_flags")
        blog_access_to_admin_enabled = flagsmith_client.get_identity_flags(
            f"org.{organisation.id}.{organisation.name}",
            traits={"organisation_id": organisation.id},
        ).is_feature_enabled("api_limiting_block_access_to_admin")
```

And then if either of the features are enabled, then the bulk update would include those files. Am I understanding this correctly?

Reply:
I was more thinking something along the lines of: 

```python
for organisation in organisations:
    flags = flagsmith_client.get_identity_flags(...)
    if flags.is_feature_enabled("api_limiting"):
        fields_to_update = json.looads(flags.get_feature_value("api_limiting"))["fields_to_update"]
        for field, value in fields_to_update.items():
            setattr(organisation, field, value)
        organisation.save()
```

... but maybe that's a bit over complicated / error prone. Perhaps your solution is better (although we should at least make sure that we separate out the call to get the flags and call `is_feature_enabled()` on the resultant Flags object rather than calling `get_identity_flags` twice).

Reply:
I think the biggest deciding factor should be how onerous it will be to update the flags once we're ready in production. My gut is slightly in favour of using individual flags over the feature values because it seems simpler to set them to enabled for the identity flags, but that may just be an incorrect understanding of how much time it will take to individually set the fields to update in your method. 

If you were in change of the Flagsmith flags for this change which approach would be more efficient?

Reply:
I think you're right tbh, individual flags are easier to manage.

Reply:
@zachaysan I think there's still work to be done here?

Reply:
Ok I've added the flagsmith flags and updated how the update logic works.